File: bot/bot.go
package bot

import (
	"log"
	"sync"
	"sync/atomic"

	"github.com/gempir/go-twitch-irc/v4"
	"github.com/gopxl/beep/v2"
)

type Bot struct {
	Client         *twitch.Client
	Channel        string
	soundsBuffer   map[string]*beep.Buffer
	mtx            sync.RWMutex
	fileMtx        sync.Mutex
	queue          []beep.Streamer
	queueEnabled   bool
	queueIsPlaying bool
	speakerIsMuted bool
	erIsOn         bool
	ttsLanguages   map[string]string
	ttsCounter     atomic.Uint64
	volume         int
}

func New(channel string, soundsBuffer map[string]*beep.Buffer, ttsLanguages map[string]string, volume int) *Bot {
	b := &Bot{
		Client:         twitch.NewAnonymousClient(),
		Channel:        channel,
		soundsBuffer:   soundsBuffer,
		queue:          make([]beep.Streamer, 0, 50),
		queueEnabled:   false,
		queueIsPlaying: false,
		speakerIsMuted: false,
		erIsOn:         true,
		ttsLanguages:   ttsLanguages,
		volume:         volume,
	}
	return b
}

func (b *Bot) IsMuted() bool {
	b.mtx.RLock()
	defer b.mtx.RUnlock()
	return b.speakerIsMuted
}

func (b *Bot) SetMuted(muted bool) {
	b.mtx.Lock()
	defer b.mtx.Unlock()
	b.speakerIsMuted = muted
}

func (b *Bot) IsQueueEnabled() bool {
	b.mtx.RLock()
	defer b.mtx.RUnlock()
	return b.queueEnabled
}

func (b *Bot) SetQEnabled(enabled bool) {
	b.mtx.Lock()
	defer b.mtx.Unlock()
	b.queueEnabled = enabled
}

func (b *Bot) printState() {
	b.mtx.RLock()
	defer b.mtx.RUnlock()
	log.Printf("Status -> Mute: %t | Queue: %t | Er: %t | Volume: %d", b.speakerIsMuted, b.queueEnabled, b.erIsOn, b.volume)
}

File: bot/Bot.cs
namespace TwitchIrcMinimal;

public class Bot
{
    public string Channel { get; }
    public int Volume { get; set; }
    public bool IsMuted { get; private set; }
    private bool _queueEnabled;
    public bool QueueEnabled
    {
        get => _queueEnabled;
        set { _queueEnabled = value; QueueChanged?.Invoke(); }
    }
    public event Action? QueueChanged;
    public bool ErEnabled { get; set; } = true;

    public AudioEngine Engine { get; }
    public Player Player { get; }
    public Dictionary<string, string> TtsLanguages { get; }
    private long _ttsCounter;

    public Bot(string channel, int volume, Dictionary<string, string> ttsLanguages)
    {
        Channel = channel;
        Volume = Math.Clamp(volume, 0, 200);
        TtsLanguages = ttsLanguages;
        Engine = new AudioEngine();
        Player = new Player(Engine);
        Engine.PlaybackFinished += () => Player.OnPlaybackFinished();
    }

    public void SetMuted(bool muted)
    {
        IsMuted = muted;
        if (muted)
        {
            Engine.Stop();
            Player.Stop();
        }
    }

    public string NextTtsKey() => $"tts_temp_{Interlocked.Increment(ref _ttsCounter)}";
}
